internal/infra/table/sql_extractor.go

```diff
@@ -128,3 +145,89 @@ func (e *SQLExtractor) Count(tableName string) (int, *table.Error) {
 
 	return count, nil
 }
+
+func ColumnInfo(db *sql.DB, tableName string) ([]table.Column, error) {
+	// Execute query to fetch column information
+	query := "SELECT * FROM "
+	query += tableName
+	query += " LIMIT 0"
+	rows, err := db.Query(query)
+	if err != nil {
+		return []table.Column{}, err
```

Comment:
I think it would be safer to log a warning here ("cannot scan columns for table <tablename>") and do not return the error.